spec/requests/chatfuel_spec.rb

```diff
@@ -8,16 +8,54 @@
     end
 
     describe "/chatfuel/:topic" do
-      let(:topic_id) { 1 }
-      let(:url) { "/chatfuel/#{topic_id}" }
+      let(:topic_name) { "milk_quality" }
+      let(:url) { "/chatfuel/#{topic_name}" }
       let(:chatbot_channel)   { Channel.chatbot }
       let(:report)            { Report.current }
 
+      describe 'topic doesn\'t exist' do
+        it { is_expected.to have_http_status(404) }
+      end
+
       describe 'text component doesn\'t exist' do
         before { create(:topic, id: 1, name: "milk_quality") }
         it { is_expected.to have_http_status(404) }
       end
 
+      describe 'text compoent with wrong topic' do
+        let(:topic) { create(:topic, id: 2, name: "milk_quantity") }
+
+        before do
+          create(
+            :text_component,
+            report: report,
+            topic: topic,
+            channels: [chatbot_channel],
+            main_part: "The main part",
+            id: 1
+          )
+        end
+
+        it { is_expected.to have_http_status(404) }
+      end
+
+      describe 'text compoent with wrong channel' do
+        let(:topic) { create(:topic, id: 1, name: "milk_quality") }
+
+        before do
+          create(
+            :text_component,
+            report: report,
+            topic: topic,
+            channels: [Channel.sensorstory],
+            main_part: "The main part",
+            id: 1
+          )
+        end
+
+        it { is_expected.to have_http_status(404) }
+      end
```

Comment:
Please add a test in wich you
1. Create text component for topic `adorable_kitten`
2. Call the controller for topic `adorable_kitten`
3. Receive the expected text component